docs: restructure docs sidebar (#1796)
urbica/martin: docs/src/run-with-reverse-proxy.md

# Reverse Proxys

Martin can run without a reverse proxy.

> Doing so has a few downsides though:

Reviewer: ```suggestion
Doing so has a few downsides:
```

> - Martin by itself does not handle certificates => you won't be able to use HTTPS.

Reviewer: ```suggestion
- Martin does not support HTTPS connections (TLS termination).
```

> - We would serve to all requests going to the port Martin is running on

Reviewer: this is not very clear

Author: True, how about

```suggestion
- We do not check `HOST`-headers - we just serve on a port.
  This means anybody can point their dns record to your server and serve to all requests going to the port Martin is running on.
  Using a reverse proxy makes this abuse obvious.
```

> - Martin only has one-size-fit-all caching. If you need more advanced caching options, you can use a reverse proxy like Nginx or Apache and cache for example only tiles from z10 and up.

Reviewer: ```suggestion
- Martin only supports a simple in-memory caching. If you need more advanced caching options, you can use a reverse proxy like Nginx, Varnish, or Apache with custom rules. For example, you may choose to only cache zoom 0..10.
```

Author: ```suggestion
- Martin only supports a simple in-memory caching.
  If you need more advanced caching options, you can use a reverse proxy like [Nginx](https://nginx.org/), [Varnish](https://varnish-cache.org/), or [Apache](https://httpd.apache.org/) with custom rules.
  For example, you may choose to only cache zoom 0..10.
```

> - You likely want more than one thing running on the same server, such as a web server.

Reviewer: ```suggestion
- You may need to host more than just tiles at a single domain name.
```

> - We don't handle subpath routing. (e.g. serving at `/tiles`)

Reviewer: ```suggestion
- Martin has a fixed public API, but your site may require a different structure, e.g. serving tiles with from a sub-path like `/tiles/source/z/x/y`.
```
